### PR TITLE
[MAINT] use tmp_path in glm reporter

### PR DESCRIPTION
**Pull request by Remi-Gau**
- relates to https://github.com/nilearn/nilearn/issues/3726
- extracted from #3891
- ~~blocked by https://github.com/nilearn/nilearn/pull/3892&~~

### CODE REVIEW

File: nilearn/reporting/tests/test_glm_reporter.py [hunk 1]
```diff
@@ -282,35 +271,32 @@ def test_plot_contrasts():
...
+        contrast,
+        plot_type="glass",
+        height_control=None,
+        min_distance=15,
+        alpha=0.001,
+        threshold=2.78,
+    )
 
-        del mask, flm, fmri_data, masker
+    report_iframe = report_flm.get_iframe()
```
Comment (ymzayek) [inline nilearn/reporting/tests/test_glm_reporter.py:300]: and here

File: nilearn/reporting/tests/test_glm_reporter.py [hunk 2]
```diff
@@ -27,76 +26,68 @@
...
-        report_iframe
-        del mask, flm, fmri_data
+    """
+    catches & raises UnicodeEncodeError in HTMLDocument.get_iframe()
+    in case certain unicode characters are mishandled,
+    like the greek alpha symbol.
+    """
+    report_iframe = report_flm.get_iframe()
+    # So flake8 doesn't complain about not using variable (F841)
+    report_iframe
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/reporting/tests/test_glm_reporter.py:64]: yeah the comment made think that flake8 would reset the universe if I removed it 
let me try removing it to see what happens
worst case scenario... Well it was nice knowing you

[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/reporting/tests/test_glm_reporter.py]: we are all still here and it seems that the comments were lie... again...

**Review by ymzayek — APPROVED** [review]
Thx!

**Pull request merged**